### PR TITLE
Adding implementation of badge expiration overlay

### PR DESCRIPTION
**Pull request by ramabit**
Screenshot 2023-07-03 at 17 59 37
Screenshot 2023-07-03 at 17 59 47
Screenshot 2023-07-03 at 17 59 06
Screenshot 2023-07-03 at 17 59 20
Screenshot 2023-07-03 at 17 59 25
Screenshot 2023-07-03 at 17 59 30

### CODE REVIEW

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        {props.badge}
+      </Box>
+      <OverlayContainer width={badgeWidth} height={badgeHeight}>
+        {/* time left rectangle */}
+        <Box>
+          <TimeLeftContainer width={timeToExpireWidth} background={props.overlayBackgroundColor || colors.white}>
+            <Typography
+              variant={'h2'}
+              component={'h2'}
+              fontSize={'64px !important'}
```
Comment (FedeMadoery) [inline src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx:68]: `fontSize` shouldn't be fixed, in the small size badges the text is too big. We must use the default configs for h2 or some other font variant that we already have configured so you don't need to do a custom one for this.
image

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+              sx={{ textAlign: 'center' }}
+              color={props.overlayTextColor || colors.black}
+              marginBottom={0.5}
+            >
+              {props.timeLeft?.quantity}
+            </Typography>
+            <Typography
+              textTransform={'uppercase'}
+              variant="dAppTitle3"
+              fontSize={'20px !important'}
```
Comment (FedeMadoery) [inline src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx:79]: Same here

**Review by FedeMadoery — CHANGES_REQUESTED** [review]

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx (hunk 1 above)
Comment (ramabit) [reply src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx]: There I updated the sizes using the same as for the other overlay. Unfortunately, there is one combination of sizes that it is not in the theme (dAppTitle2 with size 16)

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx (hunk 2 above)
Comment (ramabit) [reply src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx]: please check the new sizes

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  })
+
+  return (
+    <Box>
+      <Box ref={badgeRef} sx={{ width: 'fit-content', height: 'fit-content' }}>
+        {props.badge}
+      </Box>
+      <OverlayContainer width={badgeWidth} height={badgeHeight}>
+        {/* time left rectangle */}
+        <Box>
```
Comment (FedeMadoery) [inline src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx:64]: picky: This box maybe is not needed

**Review by FedeMadoery — APPROVED** [review]

File: src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx (hunk 3 above)
Comment (ramabit) [reply src/components/overlays/TimeToExpireBadgeOverlay/TimeToExpireBadgeOverlay.tsx:64]: let me check..

**Pull request merged**